### PR TITLE
Fixing some issues with UserProfile.jsx and Profile.jsx

### PR DESCRIPTION
**Pull request by gbowne1**
The UserProfile.jsx did not have very good code in it so I added code that will be a bit more useful.  This should make it easier to both complete fix the UserProfile.jsx

I added prop-types, useNavigate, useState, useEffect, and persisted some LocalStorage code.  I was on the fence about using LocalStorage for user profile stuff.

I will take any ideas from here.

The CSS will need to be fixed as well.

The component 404's now but will be a lot easier to fix and complete. I think we can agree that the original compnent code was not very good

### CODE REVIEW

**Review by pawel975 — APPROVED** [review]
I wrote down few suggestions but approve PR because possible changes are not crucial.

File: src/pages/Profile/Profile.jsx [hunk 1]
```diff
@@ -1,39 +1,42 @@
-import React, { useState } from 'react';
-import PropTypes from "prop-types";
-import CloseButton from "../../components/CloseButton/CloseButton";
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
```
[the remark this answers is not in the document]
Comment (gbowne1) [reply src/pages/Profile/Profile.jsx:2]: We should have a prettier config in the project then.

File: src/pages/UserProfile/UserProfile.css [hunk 2]
```diff
@@ -63,4 +63,25 @@ h3 {
     padding-top: 19px;
     height: 45px;
     width: 180px;
-}
\ No newline at end of file
+}
```
[the remark this answers is not in the document]
Comment (gbowne1) [reply src/pages/UserProfile/UserProfile.css:66]: I wasnt worried about CSS styling for this.  It was already a bit of a mess.

I have a funny feeling that user profile is going to be a pretty big file. 

We also have Profile.jsx

**Comment by pawel975** [comment]
> I get redirected to /login when ever I navigate to /user-profile or /profile.
> 
> Here is a screenshot:
> 
> image

It's because code try to get data from Local storage under "user" key, that's not generated anywhere. To check how it works you should hard code object with "user" key which contains "name" and "email" properties.

**Comment by manuel12** [comment]
> > I get redirected to /login when ever I navigate to /user-profile or /profile.
> > Here is a screenshot:
> > image
> 
> It's because code try to get data from Local storage under "user" key, that's not generated anywhere. To check how it works you should hard code object with "user" key which contains "name" and "email" properties.

Will do that, thx.

**Comment by gbowne1** [comment]
This will also need some CSS changes too.  I was more interested in getting some good base logic down.  This component will be pretty heavy, and may need to be broken into some smaller sub components.  We can work on this together.

I also forgot to add back the handlers for closing the component/panel, and the theme state.   This wil need to be added.

After the merge I (or anyone) can go back through and add some initial stuff to the body of the component.

@manuel12 @pawel975

**Comment by gbowne1** [comment]
I made a cleanup push to my main without a PR, so we can get some more work done on this.  
Mainly was re-Linting code with the new rule sets in place.

**Comment by manuel12** [comment]
> This will also need some CSS changes too. I was more interested in getting some good base logic down. This component will be pretty heavy, and may need to be broken into some smaller sub components. We can work on this together.
> 
> I also forgot to add back the handlers for closing the component/panel, and the theme state. This wil need to be added.
> 
> After the merge I (or anyone) can go back through and add some initial stuff to the body of the component.
> 
> @manuel12 @pawel975

Will review this today.

**Review by manuel12 — APPROVED** [review]
Looks good now, we can update it later in the future. 
I also believe this should be broken into smaller component, we need to figure out what
we want here. I think is best to start as simple as possible and the improve.

**Pull request merged**